pulsar-client-go/pulsar/c_producer.go

```diff
@@ -156,9 +156,7 @@ func createProducerAsync(client *client, schema Schema, options ProducerOptions,
 		C._pulsar_producer_configuration_set_message_router(conf, savePointer(&options.MessageRouter))
 	}
 
-	if options.Batching {
-		C.pulsar_producer_configuration_set_batching_enabled(conf, cBool(options.Batching))
-	}
+	C.pulsar_producer_configuration_set_batching_enabled(conf, cBool(options.Batching))
```

Comment:
The downside for this is that batching will be disabled by default on Go (unless we make the flag to be a pointer), because there's no way to set `Batching = true` by default. 

This is something we can fix with the API beak in the native go client.